### PR TITLE
CC-2601 Add S3 Append-Late-Data option to configs

### PR DESCRIPTION
**Pull request by cyrusv**

### CODE REVIEW

File: kafka-connect-s3/docs/configuration_options.rst [hunk 1]
```diff
@@ -33,13 +33,20 @@ Connector
...
-  Width to zero pad offsets in store's filenames if offsets are too short in order to provide fixed width filenames that can be ordered by simple lexicographic sorting.
+  Width to zero-pad offsets in store's filenames if offsets are too short in order to provide fixed-width filenames that can be ordered by simple lexicographic sorting.
 
   * Type: int
   * Default: 10
   * Valid Values: [0,...]
   * Importance: low
 
+``append.late.data``
+  If true, the partitions will append data from previous (already commited) partitions into the current, open partition. If false, late data intended for closed partitions will be rolled into files marked with the partition indicated by the extracted time, even if this createsmany small files.
```
Comment (sansanichfb) [inline kafka-connect-s3/docs/configuration_options.rst]: Nit: createsmany-> creates many

Comment (kkonstantine) [inline kafka-connect-s3/docs/configuration_options.rst]: the "already committed" (typo) part is not correct. You might haven't committed any records in previous partitions. Late data is just late. 

Also, not sure I understand "the partitions will append". You mean the connector?

Similar to the first comment, I'd call "closed partitions" "earlier partitions" 

typo: createsmany

File: kafka-connect-s3/docs/configuration_options.rst [hunk 2]
```diff
@@ -33,13 +33,20 @@ Connector
...
-  Width to zero pad offsets in store's filenames if offsets are too short in order to provide fixed width filenames that can be ordered by simple lexicographic sorting.
+  Width to zero-pad offsets in store's filenames if offsets are too short in order to provide fixed-width filenames that can be ordered by simple lexicographic sorting.
 
   * Type: int
   * Default: 10
   * Valid Values: [0,...]
   * Importance: low
 
+``append.late.data``
+  If true, the partitions will append data from previous (already commited) partitions into the current, open partition. If false, late data intended for closed partitions will be rolled into files marked with the partition indicated by the extracted time, even if this creates many small files.
```
Comment (wicknicks) [inline kafka-connect-s3/docs/configuration_options.rst]: do we mean `"tasks" will append data from previous (already commited) partitions into the current...`?

**Comment by cyrusv** [comment]
Now this just pushes up the changes already discussed in the common repo

You will note the target branch is 4.0.x since 3.2.x had so many incompatible changes with respect to confluent core

File: kafka-connect-s3/docs/configuration_options.rst [hunk 3]
```diff
@@ -39,6 +39,20 @@ Connector
   * Default: 1000
   * Importance: low
 
+``enhanced.avro.schema.support``
```
Comment (kkonstantine) [inline kafka-connect-s3/docs/configuration_options.rst]: Why are we adding those? Where they missing or they are just carried over?

Comment (kkonstantine) [reply kafka-connect-s3/docs/configuration_options.rst]: In this case, they need to go on a separate PR that targets the appropriate `-post` branch

Comment (cyrusv) [reply kafka-connect-s3/docs/configuration_options.rst]: They were missing  and autogenerated -- However the testing against them is weak so I'm not 100% sure they are implemented correctly. Do you have any additional insight? Looks like you added it last october

**Pull request merged**